code/modules/antagonists/eldritch_cult/eldritch_magic.dm

```diff
@@ -0,0 +1,395 @@
+	hand_path = /obj/item/melee/touch_attack/mad_touch
+	school = "evocation"
+	charge_max = 150
+	clothes_req = FALSE
+	invocation = "OP'N Y'R M'ND"
+	invocation_type = "whisper"
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "ash_shift"
+	action_background_icon_state = "bg_ecult"
+
+
+/obj/item/melee/touch_attack/mad_touch
+	name = "Ash Leech"
+	desc = "A sinister looking aura that distorts the flow of reality around it."
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	catchphrase = "M'DNESS"
+
+/obj/item/melee/touch_attack/mad_touch/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	..()
+	if(ishuman(target))
+		var/mob/living/carbon/human/tar = target
+		if(tar.anti_magic_check())
+			tar.visible_message("<span class='danger'>Spell bounces off of [target]!</span>","<span class='danger'>The spell bounces off of you!</span>")
+			return
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		if(!C.mind.has_antag_datum(/datum/antagonist/ecult))
+			SEND_SIGNAL(C, COMSIG_ADD_MOOD_EVENT, "gates_of_mansus", /datum/mood_event/gates_of_mansus)
+		return
+
+/obj/effect/proc_holder/spell/pointed/ash_final
+	name = "Nightwatcher's Rite"
+	desc = "Powerful spell that releases 5 streams of fire away from you."
+	school = "transmutation"
+	invocation = "F'RE"
+	invocation_type = "whisper"
+	charge_max = 300
+	range = 15
+	clothes_req = FALSE
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "ash_shift"
+	action_background_icon_state = "bg_ecult"
+
+/obj/effect/proc_holder/spell/pointed/ash_final/cast(list/targets, mob/user)
+	for(var/X in targets)
+		var/T
+		T = line_target(-25, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+		T = line_target(10, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+		T = line_target(0, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+		T = line_target(-10, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+		T = line_target(25, range, X, user)
+		INVOKE_ASYNC(src, .proc/fire_line, user,T)
+	. = ..()
+
+/obj/effect/proc_holder/spell/pointed/ash_final/proc/line_target(offset, range, atom/at , atom/user)
+	if(!at)
+		return
+	var/angle = ATAN2(at.x - user.x, at.y - user.y) + offset
+	var/turf/T = get_turf(user)
+	for(var/i in 1 to range)
+		var/turf/check = locate(user.x + cos(angle) * i, user.y + sin(angle) * i, user.z)
+		if(!check)
+			break
+		T = check
+	return (getline(user, T) - get_turf(user))
+
+/obj/effect/proc_holder/spell/pointed/ash_final/proc/fire_line(source, list/turfs)
+	var/list/hit_list = list()
+	for(var/turf/T in turfs)
+		if(istype(T, /turf/closed))
+			break
+		new /obj/effect/hotspot(T)
+		T.hotspot_expose(700,50,1)
+		for(var/mob/living/L in T.contents)
+			if(L.anti_magic_check())
+				L.visible_message("<span class='danger'>Spell bounces off of [L.real_name]!</span>","<span class='danger'>The spell bounces off of you!</span>")
```

Comment:
if it bounces off of them, could you make the line of effect of this spell start going back the other direction or something?

also, you might want to do the hotspot_expose() thing AFTER checking to see if anyone on the turf has antimagic, so that the person with antimagic doesn't just get set on fire anyway

Reply:
:agony: good idea